devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,551 @@
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def auto() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_AUTO),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def cool() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_COOL),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def heat() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_HEAT),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def ping() {
+	log.debug "ping"
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh"
+	getRefreshCommands()
+}
+
+def getRefreshCommands() {
+	def refreshCommands = []
+
+	refreshCommands += zigbee.readAttribute(RELATIVE_HUMIDITY_CLUSTER, RELATIVE_HUMIDITY_MEASURED_VALUE)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_HUMIDITY)
+
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT_UNOCCUPIED)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT_UNOCCUPIED)
+
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_SPEED)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE) // formerly THERMOSTAT MODE: 0x001C
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, 0x0650)	//occupancy command
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_EFFECTIVE_OCCUPANCY)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_CLUSTER, OCCUPANCY)
+	refreshCommands += zigbee.readAttribute(THERMOSTAT_UI_CONFIGURATION_CLUSTER, TEMPERATURE_DISPLAY_MODE)
+
+	refreshCommands
+}
+
+def configure() {
+	log.debug "Configuration"
+
+	if (!childDevices) {
+		createChildThermostat()
+	}
+
+	def configurationCommands = []
+	// todo: check if following binding is necessary here
+	configurationCommands += "zdo bind 0x${device.deviceNetworkId} 1 0xA 0x201 {${device.zigbeeId}} {}"
```

Comment:
I think `zigbee.addBinding(_CLUSTER_)` might also work here.

Reply:
Since binding is added in zigbee.configureReporting() method those binding commands probably  will not be needed here, but i will take another look at this (remove those commands first and check if all works correctly)

Reply:
I think @PKacprowiczS  is correct.

Reply:
Yes, `zigbee.configureReporting` has a call to `zigbee.addBinding` as part of it. So in this case, clusters 0x201 and 0x405 will be handled with all of the `zigbee.configureReporting(THERMOSTAT_CLUSTER` and `zigbee.configureReporting(RELATIVE_HUMIDITY_CLUSTER` calls below (in fact, 13 times for the thermostat cluster). So these calls are superfluous.

Reply:
Yes, I agree. As expected, I removed those zdo bind commands.